pkg/monitoring/rules/recordingrules/operator.go

```diff
@@ -0,0 +1,34 @@
+/*
+Copyright 2024 The KubeVirt Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package recordingrules
+
+import (
+	"github.com/machadovilaca/operator-observability/pkg/operatormetrics"
+	"github.com/machadovilaca/operator-observability/pkg/operatorrules"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+var operatorRecordingRules = []operatorrules.RecordingRule{
+	{
+		MetricsOpts: operatormetrics.MetricOpts{
+			Name: "kubevirt_memory_delta_from_requested_bytes",
+			Help: "The delta between the pod with highest memory working set or rss and its requested memory for each container, virt-controller, virt-handler, virt-api and virt-operator.",
+		},
+		MetricType: operatormetrics.GaugeType,
+		Expr:       intstr.FromString("label_replace(topk by(container)(1,max by(container, namespace, node)(container_memory_working_set_bytes{container=~\"virt-controller|virt-api|virt-handler|virt-operator\"}  - on(pod) group_left(node) (kube_pod_container_resource_requests{ container=~\"virt-controller|virt-api|virt-handler|virt-operator\",resource=\"memory\"}))), \"source\", \"memory_working_set\", \"\", \"\") or\nlabel_replace(topk by(container)(1,max by(container, namespace, node)(container_memory_rss{container=~\"virt-controller|virt-api|virt-handler|virt-operator\"}  - on(pod) group_left(node) (kube_pod_container_resource_requests{ container=~\"virt-controller|virt-api|virt-handler|virt-operator\",resource=\"memory\"}))), \"source\", \"memory_rss\", \"\", \"\")"),
```

Comment:
I try to do that on a real cluster! it's not working, It shows only the one register latest

Reply:
@simonpasquier That was my original approach, But Aviv tested and from what she checked it only stores the last record, not both..

Reply:
it should :tm: work though if the label sets are different between the 2 recording rules.

Reply:
We did you different labels. 
@avlitman Can you please verify this again?

Reply:
interesting, I didn't try with different labels actually. I will test it.

Reply:
Tested it and it's not working.. 

the rules:
![Screenshot from 2024-04-03 13-54-00](https://github.com/kubevirt/kubevirt/assets/64130977/473ceaf5-ec34-4352-9370-95fee2ca6283)


but it shows only records for rss...
![Screenshot from 2024-04-03 13-53-07](https://github.com/kubevirt/kubevirt/assets/64130977/58da5000-2c08-4e8d-b535-dcb6a6921580)

Reply:
I think it's good as is. This is not a heavy query I tested it and it take less then 1 sec to execute.

Reply:
Ok, I did a quick experiment and sorry but my suggestion wasn't 100% correct. You need to add a distinctive label to the recording rules to avoid overwriting one with the other.

```
        - expr: 'label_replace(topk by(pod,container)(1,max by(container, namespace, node,pod)(container_memory_rss{container=~"prometheus"} - on(pod) group_left(node) (kube_pod_container_resource_requests{ container=~"prometheus",resource="memory"}))), "source", "memory_rss", "", "")'
          labels:
            origin: rss
          record: prometheus_memory_delta_from_requested_bytes
        - expr: 'label_replace(topk by(pod,container)(1,max by(container, pod,namespace, node)(container_memory_working_set_bytes{container=~"prometheus"} - on(pod) group_left(node) (kube_pod_container_resource_requests{ container=~"prometheus",resource="memory"}))), "source", "memory_working_set", "", "")'
          labels:
            origin: working_set
          record: prometheus_memory_delta_from_requested_bytes
```

Another important information: `source` is a problematic label because it's used by ROSA OpenShift to identify data plane vs. control plane metrics (I realize this during testing). I'd advise to use something else to avoid further complications.

Reply:
@simonpasquier Changed the label to reason.
Regarding test I'm not sure what you mean I tested it manually, with cluster-up and cluster-sync.

Reply:
Hope this pr can be approved now (:

Reply:
I tested it again and it's not working even with the labels you suggested.. 
can we please approve this pr?

Reply:
I understand why it's not possible for kubevirt repos, since in  [github.com/machadovilaca/operator-observability](http://github.com/machadovilaca/operator-observability) we make sure no duplicates rules/metrics so test will not overwrite. cdi asked for this.

Reply:
we can add a flag to allow duplicates

Reply:
@machadovilaca I don't think we should have this check. We can have duplicate rules with the same name.